src/app/js/lib/components/ButtonWithDialogForm.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import translate from 'redux-polyglot/translate';
+import compose from 'recompose/compose';
+import withHandlers from 'recompose/withHandlers';
+import { submit as submitAction } from 'redux-form';
+import FlatButton from 'material-ui/FlatButton';
+import IconButton from 'material-ui/IconButton';
+import classnames from 'classnames';
+import CircularProgress from 'material-ui/CircularProgress';
+
+import ButtonWithStatus from './ButtonWithStatus';
+import { polyglot as polyglotPropTypes } from '../../propTypes';
+import ButtonWithDialog from './ButtonWithDialog';
+
+export const ButtonWithDialogFormComponent = ({
```

Comment:
loose the `Component` suffix

Reply:
I will replace it with a Pure suffix, you do not name two thing the same, if it is not the case.